app/models/shipit/deploy.rb

```diff
@@ -283,7 +283,8 @@ def update_last_deploy_time
     end
 
     def update_latest_deployed_ref
-      stack.update_latest_deployed_ref
+      return unless previous_changes.include?(:status)
+      stack.update_latest_deployed_ref if previous_changes[:status].include? 'success'
```

Comment:
`previous_changes[:status].last == 'success'`?

Though I don't think it really matters since the job has its own filter anyways.

Reply:
I think it's worth doing. It avoids enqueueing a job unnecessarily. There's few side-effects from enqueueing now, but you never know in the future, and it's more technically correct to have that guard here anyways.